backend/src/nodes/image_util_nodes.py

```diff
@@ -5,80 +5,98 @@
 
 import cv2
 import numpy as np
+import numpy.typing as npt
 from sanic.log import logger
 
 from .categories import IMAGE_UTILITY
 from .node_base import NodeBase
 from .node_factory import NodeFactory
 from .properties.inputs import *
 from .properties.outputs import *
-from .utils.image_utils import alpha_overlay, calculate_ssim
+from .utils.image_utils import blend_images, calculate_ssim
 from .utils.pil_utils import *
 from .utils.utils import get_h_w_c
 
+ndarray32 = npt.NDArray[np.float32]
 
-@NodeFactory.register("chainner:image:overlay")
-class ImOverlay(NodeBase):
-    """OpenCV transparency overlay node"""
+
+@NodeFactory.register("chainner:image:blend")
+class ImBlend(NodeBase):
+    """Blending mode node"""
 
     def __init__(self):
         """Constructor"""
         super().__init__()
-        self.description = "Overlay transparent images on base image."
+        self.description = """Blends overlay image onto base image using 
+            specified mode and opacities."""
         self.inputs = [
-            ImageInput("Base").with_id(0),
+            ImageInput("Base"),
             SliderInput(
                 "Base Opacity",
                 maximum=100,
                 default=100,
                 step=0.1,
                 controls_step=1,
                 unit="%",
-            ).with_id(5),
-            ImageInput("Overlay A").with_id(1),
-            SliderInput(
-                "Opacity A",
-                maximum=100,
-                default=50,
-                step=0.1,
-                controls_step=1,
-                unit="%",
-            ).with_id(2),
-            ImageInput("Overlay B").make_optional().with_id(3),
+            ),
+            ImageInput("Overlay").make_optional(),
```

Comment:
Why is this optional? If this image is not given, then the node is a noop (expect for the opacity slider).

Reply:
Because if it's not optional, the node won't run with base only. It is desirable for it to run with base only because a) it could simply be used to adjust image opacity and b) in case a user wants to pass an image through while testing but doesn't want to delete the node.

Reply:
> it could simply be used to adjust image opacity

That's not the primary function of this node, and if people really want to repurpose the node, they can just pass in the same image as top and bottom and set the top opacity to 0.

> in case a user wants to pass an image through while testing but doesn't want to delete the node.

Just set the opacity of the top layer to 0.

Reply:
That just makes noop more convoluted to achieve, so I don't really see the purpose behind doing that. We have a lot of nodes with noop conditions, all of which I just see as a temporary solutions to our lack of proper node disabling/passthrough. If we can make that work as a feature, we can get rid of all the noop conditions in various nodes.